app/javascript/dashboard/routes/dashboard/settings/inbox/settingsPage/ConfigurationPage.vue

```diff
@@ -164,6 +193,60 @@ export default {
         this.showAlert(this.$t('INBOX_MGMT.EDIT.API.SUCCESS_MESSAGE'));
       }
     },
+    async updateWhatsAppInboxAPIKey() {
```

Comment:
A shorter implementation of the same function would be. **This assumes that the `provider_config` has no extra data.**

```js
async updateWhatsAppInboxAPIKey() {
  try {
    const payload = {
      id: this.inbox.id,
      formData: false
    }

    if (this.isAWhatsAppCloudChannel) {
      payload.provider_config = {
        api_key: this.whatsAppInboxAPIKey,
        ...this.inbox.provider_config,
      }
    } else if (this.is360DialogWhatsAppChannel) {
      payload.provider_config =  {
        api_key: this.whatsAppInboxAPIKey,
      }
    }

    await this.$store.dispatch('inboxes/updateInbox', payload);
    this.showAlert(this.$t('INBOX_MGMT.EDIT.API.SUCCESS_MESSAGE'));
  } catch (error) {
    this.showAlert(this.$t('INBOX_MGMT.EDIT.API.ERROR_MESSAGE'));
  }
}
```